src/azure-cli/azure/cli/command_modules/acr/check_health.py

```diff
@@ -418,28 +448,157 @@ def _check_private_endpoint(cmd, registry_name, vnet_of_private_endpoint):  # py
     else:
         raise CLIError('DNS routing verification failed')
 
+# Validate --image input
+def validate_image(cmd,
+                   registry_name,
+                   repository_name,
+                   login_server,
+                   username,
+                   password,
+                   image):
+    from .manifest import _get_v2_manifest_path
+
+    from ._docker_utils import (
+        request_data_from_registry,
+        get_manifest_authorization_header,
+        RegistryException
+    )
+
+    try:
+        request_data_from_registry(
+            http_method='get',
+            login_server=login_server,
+            path=_get_v2_manifest_path(repository_name, image),
+            username=username,
+            password=password,
+            json_payload=None,
+            file_payload=None,
+            manifest_headers=get_manifest_authorization_header(username, password),
+            params=image)
+
+    except RegistryException as e:
+        if e.status_code == 404:
```

Comment:
what is the intention behind this?

Reply:
@rosanch So intention behind this was to first check if the image data given by user was valid before calling the API to check for the blob - is it even an image that exists in the repo? If it's a 404 and requested data does not exist then we check to see if it was user error and a possible typo and run '_check_image_match' so they get a more useful error message back. If there's a better way to do this, please let me know.

Reply:
Will add a comment, thanks!